api/lib/domain/services/certification-result-service.js

```diff
@@ -133,29 +113,72 @@ async function _getTestedCompetences({ userId, limitDate, isV2Certification }) {
+    result.listChallengesAndAnswers = challengeInformation;
 
-    result.listChallengesAndAnswers = _getChallengeInformation(matchingAnswers, certificationAssessment.certificationChallenges, allPixCompetences);
     return result;
   },
 };
+
+async function _getScoreAndLevels(certificationAssessment, continueOnError) {
+  // userService.getPlacementProfile() + filter level > 0 => avec allCompetence (bug)
+  const testedCompetences = await _getTestedCompetences({
+    userId: certificationAssessment.userId,
+    limitDate: certificationAssessment.createdAt,
+    isV2Certification: certificationAssessment.isV2Certification,
+  });
+
+  // map sur challenges filtre sur competence Id - S'assurer qu'on ne travaille que sur les compétences certifiables
+  const matchingCertificationChallenges = _selectChallengesMatchingCompetences(certificationAssessment.certificationChallengesInTestOrder(), testedCompetences);
+
+  // map sur challenges filtre sur challenge Id
+  const matchingAnswers = _selectAnswersMatchingCertificationChallenges(certificationAssessment.certificationAnswersByDate, matchingCertificationChallenges);
+
+  return _getResult(matchingAnswers, matchingCertificationChallenges, testedCompetences, continueOnError);
+}
+
+async function _getChallengeInformation(certificationAssessment) {
+  // userService.getPlacementProfile() + filter level > 0 => avec allCompetence (bug)
+  const testedCompetences = await _getTestedCompetences({
+    userId: certificationAssessment.userId,
+    limitDate: certificationAssessment.createdAt,
+    isV2Certification: certificationAssessment.isV2Certification,
+  });
+
+  // map sur challenges filtre sur competence Id - S'assurer qu'on ne travaille que sur les compétences certifiables
+  const matchingCertificationChallenges = _selectChallengesMatchingCompetences(certificationAssessment.certificationChallengesInTestOrder(), testedCompetences);
+
+  // map sur challenges filtre sur challenge Id
+  const matchingAnswers = _selectAnswersMatchingCertificationChallenges(certificationAssessment.certificationAnswersByDate, matchingCertificationChallenges);
+
+  const allPixCompetences = await competenceRepository.listPixCompetencesOnly();
+  return matchingAnswers.map((answer) => {
+
+    const certificationChallengeRelatedToAnswer = certificationAssessment.certificationChallengesInTestOrder().find(
+      (certificationChallenge) => certificationChallenge.challengeId === answer.challengeId,
+    ) || {};
+
+    const competenceValidatedByCertifChallenge = allPixCompetences.find((competence) => competence.id === certificationChallengeRelatedToAnswer.competenceId) || {};
+
+    return {
+      result: answer.result.status,
+      value: answer.value,
+      challengeId: answer.challengeId,
+      isNeutralized: certificationChallengeRelatedToAnswer.isNeutralized,
+      competence: competenceValidatedByCertifChallenge.index || '',
+      skill: certificationChallengeRelatedToAnswer.associatedSkillName || '',
```

Comment:
Privillegier `??` plutot que `||` pour conserver le 0 (même si ca ne devrait pas etre possible ici)
const toto = 0;
console.log(toto || 'nope') //=> nope
console.log(toto ?? 'nope') //=> 0

```suggestion
      competence: competenceValidatedByCertifChallenge.index ?? '',
      skill: certificationChallengeRelatedToAnswer.associatedSkillName ?? '',
```